Create and run action

FalseR20 opened the pull request:
PR for issue #40

AlexandrZagorskiy commented on README.md:
```diff
@@ -776,6 +776,60 @@ wait_agent(3, question, ScKeynodes[QuestionStatus.QUESTION_FINISHED])
 
+### Prepare, call and execute action
+
+Function that allow to create action and call later.
+
+Prepare action function creates action and adds arguments and concepts.
```
Can you provide a case when this logic can be used? I didn't see any reason to create such construction in memory, and in a separate place call it

FalseR20 replied:
Action queue. I need to create and return an action node, wait for completion, and then run

AlexandrZagorskiy commented on docs/changelog.md:
```diff
@@ -5,8 +5,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+- Prepare and run action
 
-## [0.1.0]
+## [v0.1.0]
```
Is CI release workflow will work with such style? We make a release, but I don't remember detailes was there any problems with version naming and release notes

FalseR20 replied:
We had promlems with it because script didn't find `v0.1.0`

AlexandrZagorskiy commented on tests/test_utils/test_action_utils.py:
```diff
@@ -57,3 +63,11 @@ def test_call_agent(self):
             assert execute_agent({}, [], test_node_idtf)[1]
         self.server.remove_modules(module)
+
+    def test_call_action(self):
+        module = ScModuleTest()
+        self.server.add_modules(module)
+        with self.server.register_modules():
+            action_node = prepare_action({}, [])
+            assert execute_action(action_node, test_node_idtf)
+        self.server.remove_modules(module)
```
What about testing another functions?

FalseR20 replied:
Execute action has call action inside, but I think it needs to be added

FalseR20 replied:
Done

NikitaZotov commented on docs/changelog.md:
```diff
@@ -5,8 +5,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+- Prepare and run action
 
-## [0.1.0]
+## [v0.1.0]
```
If you change version tag here then you need to change in github releases and tags. So I propose to do not achange it, to provide consistency in history

FalseR20 replied:
It's already changed in release description

NikitaZotov commented on README.md:
```diff
@@ -776,6 +776,60 @@ wait_agent(3, question, ScKeynodes[QuestionStatus.QUESTION_FINISHED])
 
+### Prepare, call and execute action
+
+Function that allow to create action and call later.
```
```suggestion
Function that allows to create action and call later.
```

FalseR20 replied:
*Functions

FalseR20 replied:
Done

NikitaZotov commented on README.md:
````diff
@@ -776,6 +776,60 @@ wait_agent(3, question, ScKeynodes[QuestionStatus.QUESTION_FINISHED])
+Prepare action function creates action and adds arguments and concepts.
+
+```python
+def prepare_action(
+        arguments: Dict[ScAddr, IsDynamic],
+        concepts: List[Idtf],
+) -> ScAddr: ...
+```
+
+Then call or execute action like `call_agent` or `execute_agent`.
+Action call functions don't return action_node because it's parameter to them.
+
+```python
+def call_action(
````
Are there cases when you need to call this functions separately?

FalseR20 replied:
No, because one of them is part of another

FalseR20 commented:
Now there is no prepare_action(arguments, concepts).
I changed it on create_action(concepts) and moved arguments to call, 
because there was no ability to create action without arguments

FalseR20 commented:
I took out add_arguments from call_action
